From f81f93d9350b0179bfcfd53187fb64e331124907 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Thu, 9 Oct 2025 19:05:37 +0300
Subject: [PATCH 401/553] QObject: docs: clarify that "window system events"
 means the event loop

Fixes: QTBUG-140923
Pick-to: 6.8 6.5
Change-Id: Ia8ee47715f787847204c3addf2d2d0e85e202685
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit a7f39d070dee85784135cf0ffd935a39094b4fd9)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qobject.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/kernel/qobject.cpp b/src/corelib/kernel/qobject.cpp
index 78e946102c4..9dcbaac1ed4 100644
--- a/src/corelib/kernel/qobject.cpp
+++ b/src/corelib/kernel/qobject.cpp
@@ -1828,8 +1828,8 @@ int QObject::startTimer(int interval, Qt::TimerType timerType)
 
     A timer event will occur every \a interval until killTimer()
     is called. If \a interval is equal to \c{std::chrono::duration::zero()},
-    then the timer event occurs once every time there are no more window
-    system events to process.
+    then the timer event occurs once every time control returns to the event
+    loop, that is, there are no more native window system events to process.
 
     \include timers-common.qdocinc negative-intervals-not-allowed
 
-- 
2.51.2

